tests/eval/test_2fa_duo.py

```diff
@@ -0,0 +1,408 @@
+"""
+This set of tests is based on the manual tests found in the
+IdP Release Candidate Test Matrix Version 4.2.4 (5/27/2021) here
+https://wiki.cac.washington.edu/display/SMW/IAM+Team+Wiki
+
+2FA-1 thru 2FA-11. 2FA-8b and 2FA-10 are not yet automate-able.
+"""
+
+import pytest
+
+from webdriver_recorder.browser import Chrome
+
+from tests.helpers import CTRL_KEY, Locators, WebTestUtils
+from tests.models import ServiceProviderInstance, AccountNetid, TestSecrets
+
+from selenium import webdriver
+
+
+@pytest.fixture
+def browser() -> Chrome:
+    return Chrome()
+
+
+@pytest.fixture(scope='class')
+def class_browser() -> Chrome:
+    browser = Chrome()
+    try:
+        yield browser
+    finally:  # Even if there is an error
+        browser.close()
+
+
+@pytest.fixture
+def new_tab(browser):
+    def inner():
+        browser.execute_script("window.open('');")
+        browser.switch_to.window(browser.window_handles[-1])
+    return inner
+
+
+@pytest.fixture
+def close_tab(browser):
+    def inner():
+        browser.find_element_by_tag_name('body').send_keys(CTRL_KEY + 'w')
+    return inner
+
+
+@pytest.fixture
+def netid() -> str:
+    return AccountNetid.sptest01.value
+
+
+@pytest.fixture
+def netid2() -> str:
+    return AccountNetid.sptest02.value
+
+
+@pytest.fixture
+def netid3() -> str:
+    return AccountNetid.sptest03.value
+
+
+@pytest.fixture
+def netid4() -> str:
+    return AccountNetid.sptest04.value
+
+
+@pytest.fixture
+def netid5() -> str:
+    return AccountNetid.sptest05.value
+
+
+@pytest.fixture
+def clean_browser(browser: Chrome) -> Chrome:
+    browser.delete_all_cookies()
+    return browser
```

Comment:
(FYI only, not a request) Something to note is that the `delete_all_cookies` method does not actually delete all cookies. It only delete cookies for the domain that the browser is currently on. 

If your tests are working consistently right now, then you probably don't have any issues at the moment, just be cautious moving forward. (I just had an idea of how to handle this better but it would require changes to the recorder).

This is how we deal with it in identity-uw-web-tests: 

https://github.com/UWIT-IAM/identity-uw-web-tests/blob/master/tests/conftest.py#L352-L373

Reply:
Thanks for the tip! This is consistent with my own findings. It's in my back pocket ;)